components/example-notebook-servers/jupyter-pytorch-gaudi/Dockerfile

```diff
@@ -0,0 +1,143 @@
+ && apt-get autoremove \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# install elastic fabric adapter
+RUN apt-get update && export EFA_VERSION=1.29.0 && mkdir /tmp/efa \
+ && wget -q -O- https://efa-installer.amazonaws.com/aws-efa-installer-$EFA_VERSION.tar.gz | tar xz -C /tmp/efa \
+ && cd /tmp/efa/aws-efa-installer && ./efa_installer.sh -y --skip-kmod --skip-limit-conf --no-verify \
+ && rm -rf /tmp/efa && rm -rf /etc/ld.so.conf.d/efa.conf /etc/profile.d/efa.sh \
+ && apt-get autoremove && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# install habana packages
+RUN wget -O- https://${ARTIFACTORY_URL}/artifactory/api/gpg/key/public | gpg --dearmor -o /usr/share/keyrings/habana-artifactory.gpg \
+ && chown root:root /usr/share/keyrings/habana-artifactory.gpg \
+ && chmod 644 /usr/share/keyrings/habana-artifactory.gpg \
+ && echo "deb [signed-by=/usr/share/keyrings/habana-artifactory.gpg] https://${ARTIFACTORY_URL}/artifactory/debian jammy main" | tee -a /etc/apt/sources.list \
+ && apt-get update \
+ && apt-get install -y --no-install-recommends \
+    habanalabs-rdma-core="$VERSION"-"$REVISION" \
+    habanalabs-thunk="$VERSION"-"$REVISION" \
+    habanalabs-firmware-tools="$VERSION"-"$REVISION" \
+    habanalabs-graph="$VERSION"-"$REVISION" \
+ && apt-get autoremove --yes && apt-get clean && rm -rf /var/lib/apt/lists/* \
+ && sed --in-place "/$ARTIFACTORY_URL/d" /etc/apt/sources.list
+
+# install libfabric
+RUN wget -nv -O /tmp/libfabric-${LIBFABRIC_VERSION}.tar.bz2 https://github.com/ofiwg/libfabric/releases/download/v${LIBFABRIC_VERSION}/libfabric-${LIBFABRIC_VERSION}.tar.bz2 \
+ && cd /tmp/ && tar xf /tmp/libfabric-${LIBFABRIC_VERSION}.tar.bz2 \
+ && cd /tmp/libfabric-${LIBFABRIC_VERSION} \
+ && ./configure --prefix=$LIBFABRIC_ROOT --enable-psm3-verbs --enable-verbs=yes --with-synapseai=/usr \
+ && make -j && make install && cd / && rm -rf /tmp/libfabric-${LIBFABRIC_VERSION}.tar.bz2 /tmp/libfabric-${LIBFABRIC_VERSION}
+
+# install hccl wrapper for ofi
+RUN wget -nv -O /tmp/main.zip https://github.com/HabanaAI/hccl_ofi_wrapper/archive/refs/heads/main.zip \
+ && unzip /tmp/main.zip -d /tmp \
+ && cd /tmp/hccl_ofi_wrapper-main \
+ && make \
+ && cp -f libhccl_ofi_wrapper.so /usr/lib/habanalabs/libhccl_ofi_wrapper.so \
+ && cd / \
+ && rm -rf /tmp/main.zip /tmp/hccl_ofi_wrapper-main
+
+RUN /sbin/ldconfig
+
+USER $NB_UID
+
+# install habana pytorch and media python libraries
+RUN python3 -m pip install habana_media_loader=="${VERSION}"."${REVISION}"
+
+RUN mkdir /tmp/gaudipt \
+ && wget -q https://vault.habana.ai/artifactory/gaudi-pt-modules/${VERSION}/${REVISION}/pytorch/ubuntu${OS_NUMBER}/pytorch_modules-v${PYTORCH_VERSION}_${VERSION}_${REVISION}.tgz -O - | tar xz -C /tmp/gaudipt \
+ && cd /tmp/gaudipt && PYTHON_VERSION=3.10 bash install.sh ${VERSION} ${REVISION} \
+ && rm -rf /tmp/gaudipt
+
+ENV LD_PRELOAD=/lib/x86_64-linux-gnu/libtcmalloc.so.4
+
+RUN echo "source /etc/profile.d/habanalabs.sh" >> ~/.bashrc
+
```

Comment:
Because things are being updated in `$HOME`, you will need to add this instruction:

https://github.com/kubeflow/kubeflow/blob/f6563b293808c771db05023bccdd793528c53745/components/example-notebook-servers/rstudio/Dockerfile#L151-L157

Reply:
Thanks, I wasn't aware of this.

Reply:
I added this, but maybe it's not needed if the `.bashrc` sourcing is not used anymore?